esp-hal/src/uart.rs

```diff
@@ -897,15 +863,16 @@ where
         }
 
         if self.rx_fifo_count() > 0 {
-            Ok(fifo.read().rxfifo_rd_byte().bits())
+            let byte = crate::interrupt::free(|| fifo.read().rxfifo_rd_byte().bits());
```

Comment:
This only needs to be done on the esp32, could you `cfg_if` that accordingly, and add back the context as to why its needed, just the link to https://docs.espressif.com/projects/esp-chip-errata/en/latest/esp32/03-errata-description/esp32/cpu-subsequent-access-halted-when-get-interrupted.html is probably enough.

Reply:
Oops missed that `cfg`, will update that in the morning 😅